shell/platform/darwin/ios/platform_message_handler_ios.mm

```diff
@@ -127,8 +124,8 @@ - (void)dispatch:(dispatch_block_t)block {
   message_handlers_.erase(channel);
   if (handler) {
     message_handlers_[channel] = {
-        .task_queue = fml::scoped_nsprotocol(
-            [static_cast<NSObject<FlutterTaskQueueDispatch>*>(task_queue) retain]),
+        .task_queue =
+            fml::scoped_nsprotocol(static_cast<NSObject<FlutterTaskQueueDispatch>*>(task_queue)),
```

Comment:
This is similar to the comment on the other PR.  ARC doesn't insert an unbalanced retain at a call site and the constructor for the scoped_nsprotocol uses `__unsafe_unretained`, assuming that it is coming in retained.  So I think you need to add a retain here.  Looks like ScopedTypeRef has a second parameter that you can use to specify that you want it to retain.  See `fml::scoped_policy::OwnershipPolicy`

Reply:
This one is slightly different since the scoped_nsprotocol isn't consuming a `ns_returns_not_retained` return value.

`scoped_nsprotocol` is expected to consume a +1 object as seen here:
```c++
  explicit ScopedTypeRef(
      __unsafe_unretained T object = Traits::InvalidValue(),
      fml::scoped_policy::OwnershipPolicy policy = fml::scoped_policy::kAssume)
      : object_(object) {
    if (object_ && policy == fml::scoped_policy::kRetain) {
      object_ = Traits::Retain(object_);
    }
  }
```

In this new code, where would the extra retain come from that will match the one in the destructor for the scoped pointer?  I don't think there is one.

Reply:
Oh wait, i see it now:
```c++
#if !defined(__has_feature) || !__has_feature(objc_arc)
  explicit scoped_nsprotocol(NST object = Traits::InvalidValue(),
                             scoped_policy::OwnershipPolicy policy =
                                 scoped_policy::OwnershipPolicy::kAssume)
      : ScopedTypeRef<NST, Traits>(object, policy) {}
#else
  explicit scoped_nsprotocol(NST object = Traits::InvalidValue())
      : ScopedTypeRef<NST, Traits>(object,
                                   scoped_policy::OwnershipPolicy::kRetain) {}
#endif
```